arch/ia32/crt0.S

```diff
@@ -20,5 +20,6 @@
 .globl _start
 .type _start, %function
 _start:
-	jmp _startc
+	/* Call to ensure proper alignment. As %esp is 16 aligned at program entry. */
+	call _startc
```

Comment:
I don't get this. `call` does not align the stack in any way, it just pushes EIP (4 bytes). 16 byte alignment is a compiler requirement and not enforced by the CPU at all

Reply:
The call is here to ensure proper alignment in _startc, that is 16byte alignment prior to call. Should jmp be used _startc would be entered with rsp aligned to 16bytes, but it should be entered with (rsp -4) being 16 byte aligned